pkg/controllers/dynakube/extension/statefulset.go

```diff
@@ -0,0 +1,387 @@
+package extension
```

Comment:
I have similar task to create OTEL collector StatefulSet, so IMO it's better if we create subpackages or somehow split this file, otherwise I need to refactor you part as well.